### DISCUSSION TITLE
Shall I need to destroy the xml_document?

### DISCUSSION THREAD

**Discussion by siriusdemon** [zeux/pugixml#562] [Q&A]
Hello! 
I have tried to wrap the `xml_document` with `unique_ptr` but fail to run.  The document says
```
xml_document is the owner of the entire document structure; destroying the document destroys the whole tree.
```
But it never mentions a destroy function.  
I have read the source code but still could not figure it out.

**Comment by zeux** [comment]
unique_ptr should work; "destroying" in the manual mostly refers to automatic C++ destruction (whether managed by unique_ptr/shared_ptr/stack variable/etc.)

**Reply by siriusdemon** [reply]
Thanks a lot!
